src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,94 +53,1321 @@
+     * {@code possibleReceivers}. The response rows are configured using 
+     * {@code configureResponseRowForViewType(viewType)}. 
+     *  
+     * @see configureResponseRowForViewType
+     */
+    private List<InstructorResultsResponseRow> buildMissingResponseRowsBetweenGiverAndPossibleRecipients(
+                                                                    FeedbackQuestionAttributes question, 
+                                                                    List<String> possibleReceivers, 
+                                                                    String giverIdentifier,
+                                                                    String giverName, String giverTeam) {
+        List<InstructorResultsResponseRow> missingResponses = new ArrayList<InstructorResultsResponseRow>();
+        FeedbackQuestionDetails questionDetails = question.getQuestionDetails();
+        
+        for (String possibleRecipient : possibleReceivers) {
+            String possibleRecipientName = bundle.getFullNameFromRoster(possibleRecipient);
+            String possibleRecipientTeam = bundle.getTeamNameFromRoster(possibleRecipient);
+            
+            String textToDisplay = questionDetails.getNoResponseTextInHtml(giverIdentifier, possibleRecipient, bundle, question);
+            
+            if (questionDetails.shouldShowNoResponseText(giverIdentifier, possibleRecipient, question)) {
+                InstructorResultsModerationButton moderationButton = buildModerationButtonForGiver(
+                                                                         question, giverIdentifier, 
+                                                                         "btn btn-default btn-xs", "Moderate Response");
+                InstructorResultsResponseRow missingResponse = new InstructorResultsResponseRow(giverName, giverTeam, 
+                                                                                                possibleRecipientName, possibleRecipientTeam, 
+                                                                                                textToDisplay, true, moderationButton, true);
+                
+                missingResponse.setRowAttributes(new ElementTag("class", "pending_response_row"));
+                configureResponseRow(question, giverIdentifier, possibleRecipient, missingResponse);
+                missingResponses.add(missingResponse);
+            }
+        }
+        
+        return missingResponses;
+    }
+    
+    /**
+     * Construct missing response rows between the recipient identified by {@code recipientIdentifier} and 
+     * {@code possibleGivers}. The response rows are configured using 
+     * {@code configureResponseRowForViewType(viewType)}. 
+     *  
+     * @see configureResponseRowForViewType
+     */
+    private List<InstructorResultsResponseRow> buildMissingResponseRowsBetweenRecipientAndPossibleGivers(
+
```

Comment:
This may be slightly clearer by using `giverIdentifier` on both lines.

Or rather, maybe combine the conditions into the `if`?
```
String giverIdentifier = giverEmail;
if (question != null && question.giverType.isTeam()) {
    giverIdentifier = giverIdentifier.replace(Const.TEAM_OF_EMAIL_OWNER, "");
}
```